common/transcript.go

```diff
@@ -12,6 +13,7 @@ import (
 // See: Fiat-Shamir
 type Transcript struct {
 	state hash.Hash
+	buff  *bytes.Buffer
```

Comment:
Here's the optimization I mentioned around "buffering" FS writes until we need to pull a scalar challenge.